doc/howto/write_an_alphabet/index.md

```diff
@@ -0,0 +1,165 @@
+
+A look at the documentation of seqan3::Alphabet will reveal that it is actually a refinement of other concepts,
+more precisely seqan3::Semialphabet which in turn refines std::CopyConstructible and std::StrictTotallyOrdered.
+Let's check those:
+\snippet dna2_only_rank.cpp other_concepts
+
+You should see that your type models only the std::CopyConstructible concept.
+Let's have a look at the documentation for std::StrictTotallyOrdered. Can you guess what it describes?
+
+It describes the requirements for types that are comparable via `<`, `<=`, `>` and `>=`.
+This is useful so that you can sort a text over the alphabet, for example.
+Additionally, std::StrictTotallyOrdered is again a refinement of std::EqualityComparable,
+which requires the `==` and `!=` operators, so let's first implement those.
+\snippet dna2_equality_operator.cpp equality
+
+As you can see we chose to implement them as friend functions. You can also implement them as free functions in the
+namespace of your class, but you should avoid regular member functions, because they result in different conversion
+rules for left-hand-side and right-hand-side.
+
+\assignment{Excercise}
+Implement the inequality operator (!=) for `dna2`.
+\endassignment
+\solution
+\snippet dna2_inequality_operator.cpp inequality
+\endsolution
+
+We see that our type now models std::EqualityComparable, which is a prerequisite of std::StrictTotallyOrdered.
+
+\assignment{Excercise}
+Implement the four comparison operators and verify that your type models std::StrictTotallyOrdered.
+\endassignment
+\solution
+\snippet dna2_comparison_operators.cpp comparison
+\endsolution
+
+## Semialphabet
+
+Let's move on to the more interesting concepts. seqan3::Semialphabet constitutes the *rank interface*
+that we introduced in the [alphabet tutorial](\ref tutorial_alphabets). Have a look at the API reference again.
+Beyond the conceptional requirements, it also requires that seqan3::alphabet_size and seqan3::to_rank can be
+called on your alphabet.
+
+There are different ways to satisfy seqan3::alphabet_size and seqan3::to_rank, have a look at the respective API
+reference and also the [documentation on customisation points](\ref about_customisation).
+
+In this case we choose to implement the functionality as member functions:
+\snippet dna2_semialphabet.cpp semialphabet
+
+As you can see from the `static_assert` our dna2 alphabet now models seqan3::Semialphabet and
+seqan3::WritableSemialphabet:
+\snippet dna2_semialphabet.cpp writable_semialphabet_concept
+
+With this we can automatically use some free functions on our dna2 alphabet. Here is an example:
+\snippet dna2_semialphabet.cpp free_functions
+
+## Alphabet
+
+Now that you have a feeling for concepts, have a look at seqan3::Alphabet and seqan3::WritableAlphabet and make
+your type also model these concepts. If you keep the suggested namings, SeqAn will automatically
+expose the members to the respective global functions as we have seen above.
```

Comment:
delete the last sentence as it is a little confusing (what suggested naming, there no "global" functions, they are functions objects...)